lib/Remote/RemoteWebDriver.php

```diff
@@ -110,23 +120,45 @@ public static function create(
             $executor->setRequestTimeout($request_timeout_in_ms);
         }
 
+        // W3C
+        $parameters = [
+            'capabilities' => [
+                'firstMatch' => [$desired_capabilities->toArray()],
+            ],
+        ];
+
+        // Legacy protocol
+        if (null !== $required_capabilities && $required_capabilities_array = $required_capabilities->toArray()) {
```

Comment:
is this case really part of the legacy protocol, or should the comment placed above the next `if` instead ?

Reply:
I'm not sure... ping @OndraM